src/components/CmsZone.vue

```diff
@@ -194,27 +192,46 @@ class ZoneObserverManager {
 
 @Component({
   name: 'cms-zone',
-  components: { CmsCarousel, CmsContent, CmsInspectSheet },
+  components: { CmsCarousel, CmsContent, CmsInspectSheet, CmsObserver },
 })
 export default class CmsZone extends Vue {
   @Prop(String) zoneId!: string;
   @Prop(Object) extra!: {};
   @Prop(Object) context!: {};
 
   zoneStatus: string | null = null;
-  zoneType: string = '';
-  zoneHeader: string = '';
-  zoneFooter: string = '';
+  lastResponse: CMSZoneResponse | null = null;
   contents: Content[] = [];
-  cursor: string = '';
-  scrollable: Element | null = null;
+  observed: Element[] = [];
+  shouldShowInspectModal = false;
 
   nonce: number = 0;
   cursorLoading: boolean = false;
-  next = debounce(() => this.getNextPage(), 400);
-  observed: Element[] = [];
+  allContentLoaded: boolean = false;
 
-  shouldShowInspectModal = false;
+  next = throttle(async (observer: IntersectionObserver) => {
+    if (this.cursorLoading) {
+      return [];
+    }
+
+    if (this.allContentLoaded) {
+      return [];
+    }
+
+    this.cursorLoading = true;
+    try {
+      const results = await this.getNextPage();
+
+      await Vue.nextTick();
+      if (!results.length) {
+        this.allContentLoaded = true;
+      } else if (!observer.takeRecords().length) {
```

Comment:
what does this line do?

Reply:
It re-calculates all the intersections. So suppose the first loaded page is smaller than the screen height, we would immediately want to load the second page to fill out the screen, otherwise the intersections won't be computed again as the page is not tall enough to scroll.